loadtests/simulation.py

```diff
@@ -0,0 +1,318 @@
+import json
+import os
+import random
+import uuid
+
+from konfig import Config
+from loads.case import TestCase
+from requests.auth import HTTPBasicAuth
+
+
+ACTIONS_FREQUENCIES = [
+    ('create', 20),
+    ('create_put', 20),
+    ('batch_create', 50),
+    ('batch_create_put', 50),
+    ('batch_replace', 50),
+    ('batch_update', 50),
+    ('update', 50),
+    ('filter_sort', 60),
+    ('delete', 10),
+    ('batch_delete', 10),
+    ('poll_changes', 90),
+    ('list_deleted', 40),
+    ('batch_count', 50),
+    ('list_continuated_pagination', 80),
+]
+
+
+def build_record():
+    suffix = uuid.uuid4().hex
+    data = {
+        "name": "Mushroom {0}".format(suffix),
+        "editable": (random.randint(0, 1) == 0),
+        "size": random.randint(0, 10),
+    }
+    return data
+
+
+class SimulationLoadTest(TestCase):
+
+    def __init__(self, *args, **kwargs):
+        super(SimulationLoadTest, self).__init__(*args, **kwargs)
+
+        self.conf = self._get_configuration()
+
+        self.random_user = uuid.uuid4().hex
+        self.auth = HTTPBasicAuth(self.random_user, 'secret')
+        self.session.auth = self.auth
+        self.session.headers.update({'Content-Type': 'application/json'})
+
+        self.collection = 'psilos'
+        self.init_record()
+
+    def _get_configuration(self):
+        # Loads is removing the extra information contained in the ini files,
+        # so we need to parse it again.
+        config_file = self.config['config']
+        # When copying the configuration files, we lose the config/ prefix so,
+        # try to read from this folder in case the file doesn't exist.
+        if not os.path.isfile(config_file):
+            config_file = os.path.basename(config_file)
+            if not os.path.isfile(config_file):
+                msg = 'Unable to locate the configuration file, aborting.'
+                raise LookupError(msg)
+        return Config(config_file).get_map('loads')
+
+    def api_url(self, path):
+        url = "{0}/v0/{1}".format(self.server_url.rstrip('/'), path)
+        return url
+
+    def collection_url(self, prefix=True):
+        if prefix:
+            return self.api_url(self.collection)
+        return '/' + self.collection
+
+    def record_url(self, record_id, prefix=True):
+        return self.collection_url(prefix) + '/%s' % record_id
+
+    def init_record(self, *args, **kwargs):
```

Comment:
not sure why this is in a separate method: it's not called by anything else than the constructor.